tlair/src/main/scala/at/forsyte/apalache/tla/lir/Builder.scala

```diff
@@ -3,245 +3,467 @@ package at.forsyte.apalache.tla.lir
 import at.forsyte.apalache.tla.lir.oper._
 import at.forsyte.apalache.tla.lir.values.TlaBoolSet
 import at.forsyte.apalache.tla.lir.values._
-import at.forsyte.apalache.tla.lir.UntypedPredefs.untyped
 
 /**
- * A builder for TLA expressions.
+ * The base class of expressions that are produced by the builder. These expressions are to be further translated
+ * by the code that knows how to treat types. E.g., see TypedPredefs and UntypedPredefs.
+ */
+sealed trait BuilderEx {
+
+  /**
+   * Wrap the expression block with an alias, which can be used for type assignments.
+   *
+   * @param alias alias name
+   * @return a new block that wraps this one
+   */
+  def ?(alias: String): BuilderAlias = {
+    BuilderAlias(this, alias)
+  }
+}
+
```

Comment:
I wanted to keep the operator name as small as possible. See how complex the built expressions are in tests: https://github.com/informalsystems/apalache/blob/488839d9c11fa7881e9080a203690b8320a4ae64/tla-pp/src/test/scala/at/forsyte/apalache/tla/pp/TestDesugarer.scala

Reply:
I mentioned in the meeting, but IMO this would justify using the cryptic operator in tests, but not in the main body of the code, where we should be optimizing for readability and maintainability.